raster-foundry/raster-foundry
# Add and use multitiff COG trait for Landsat 8 imagery

## Overview

This PR adds a trait for `BacksplashImage`s that have their source data spread out over several tiffs. It then uses that trait to provide a new reader for Landsat 8 imagery.

### Checklist


### Demo

I was trying to figure out how to edit this to add a dumb ORLY overlay in red block text to the first frame but it took more than four minutes and didn't seem worth it

![uningested-l8](https://user-images.githubusercontent.com/5702984/64990056-9e0e2f80-d894-11e9-8458-fce37a7049c3.gif)

### Notes

~[wip] to get CI going and until I get histograms going for uningested scenes and until the flag for multitiff usage exists (`reference.conf` exists in the `db` project now so that shouldn't be too difficult)~

Reading histograms from source right now, since we don't really know what we're going to do about that. Should we add histograms to layer attributes any time someone adds a scene to a project? Cache them in memcached on the first read and just let whoever the first consumer of a scene is pay a small penalty the first time they view it? I don't know! Something to think about.

## Testing Instructions

- add an uningested landsat 8 scene to the test data Landsat 8 project
- view imagery
- you should see both scenes
- disable the flag for `BACKSPLASH_ENABLE_MULTITIFF` in the backsplash / api-server services in docker-compose
- restart servers and reload the page
- you should see what you used to expect -- only the ingested scene

Closes #4878
